lib/src/main/java/io/ably/lib/util/PlatformAgentProvider.java

```diff
@@ -0,0 +1,10 @@
+package io.ably.lib.util;
+
+public interface PlatformAgentProvider {
+    /**
+     * Creates the platform agent for agent header {@link AgentHeaderCreator}.
+     *
+     * @return Platform agent string or null if not available.
```

Comment:
I note that there's an empty implementation for Java.

Taking a quick look around, it seems to me that we must be able to query the Java runtime version. That would be very useful information to us. See [here](https://stackoverflow.com/a/2591122). It should be coded defensively, but could be helpful. I also wonder if it's also something we should query on Android too as it should still be a valid concept in that context also.

Reply:
I've added the JRE to the previously empty Java platform header :wink: https://github.com/ably/ably-java/pull/671/commits/2e870ac7a94225ada965fe5d6e962325d8b493f0

Reply:
And here's the agents file update PR https://github.com/ably/ably-common/pull/95

Reply:
🚀